pkg/service/credential/storage.go

```diff
@@ -489,8 +525,46 @@ func (cs *Storage) GetCredentialsByIssuerAndSchema(ctx context.Context, issuer s
 	return cs.getCredentialsByIssuerAndSchema(ctx, issuer, schema, credentialNamespace)
 }
 
-func (cs *Storage) GetStatusListCredentialsByIssuerAndSchema(ctx context.Context, issuer string, schema string) ([]StoredCredential, error) {
-	return cs.getCredentialsByIssuerAndSchema(ctx, issuer, schema, statusListCredentialNamespace)
+func (cs *Storage) GetStatusListCredentialsByIssuerAndSchema(ctx context.Context, statusPurpose statussdk.StatusPurpose, issuer string, schema string) ([]StoredCredential, error) {
+	keys, err := cs.db.ReadAllKeys(ctx, statusListCredentialNamespace)
+	if err != nil {
+		return nil, util.LoggingErrorMsgf(err, "could not read credential storage while searching for creds for issuer: %s", issuer)
+	}
+
+	query := "sc:" + schema + "-sp:" + string(statusPurpose)
+	var issuerSchemaKeys []string
+	for _, k := range keys {
+		if strings.Contains(k, issuer) && strings.HasSuffix(k, query) {
+			issuerSchemaKeys = append(issuerSchemaKeys, k)
+		}
+	}
+
+	if len(issuerSchemaKeys) == 0 {
+		logrus.Warnf("no credentials found for issuer: %s and schema %s", util.SanitizeLog(issuer), util.SanitizeLog(schema))
+		return nil, nil
+	}
+
+	// now get each credential by key
+	var storedCreds []StoredCredential
+	for _, key := range issuerSchemaKeys {
+		credBytes, err := cs.db.Read(ctx, statusListCredentialNamespace, key)
+		if err != nil {
+			logrus.WithError(err).Errorf("could not read credential with key: %s", key)
+		} else {
+			var cred StoredCredential
+			if err = json.Unmarshal(credBytes, &cred); err != nil {
+				logrus.WithError(err).Errorf("could not unmarshal credential with key: %s", key)
+			}
+
+			storedCreds = append(storedCreds, cred)
+		}
+	}
+
+	if len(storedCreds) == 0 {
+		logrus.Warnf("no credentials able to be retrieved for issuer: %s", issuerSchemaKeys)
+	}
```

Comment:
What's the advantage of warning here, vs doing nothing?

Reply:
As posted above: 

This is the the same for a lot of the original functions we have in this same file:
GetCredentialsBySchema, GetCredentialsBySubject etc..

The current functionality is if there are 100 creds and one fails to read it will return 99,

probably we want this to return all 100 or nothing right? I can make changes to all areas where it is like this.

Reply:
I'm inclined to make the changes, may break a few unit tests and change functionality

Reply:
Not too much, just for debugging we can see that people are querying a cred for a certain issuer that does not exist i suppose

Reply:
think it's useful to keep